azure-iot-device/samples/async-hub-scenarios/recurring_telemetry.py

```diff
@@ -0,0 +1,55 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import asyncio
+import time
+import uuid
+from azure.iot.device.aio import IoTHubDeviceClient
+from azure.iot.device import Message
+
+
+async def send_recurring_telemetry(device_client):
+    # Connect the client.
+    await device_client.connect()
+
+    # Send recurring telemetry
+    i = 0
+    while True:
+        i += 1
+        msg = Message("test wind speed " + str(i))
+        msg.message_id = uuid.uuid4()
+        msg.correlation_id = "correlation-1234"
+        msg.custom_properties["tornado-warning"] = "yes"
+        msg.content_encoding = "utf-8"
+        msg.content_type = "application/json"
+        print("sending message #" + str(i))
+        await device_client.send_message(msg)
+        time.sleep(2)
```

Comment:
This has to use `time.sleep()` since `asyncio.sleep()` has a somewhat different behavior, and turns program exit via the `KeyboardInterrupt` into a nightmare (even more than it already is annoying to deal with in the async event loop).

When I get to implementing support for async cancels, we might be able to start using `asyncio.sleep()`, not sure yet.